### PR TITLE
Docs: Fixing redirects

### LINKED ISSUE #9454: Docs: 404 pages
After publishing the `12.0` version of the docs, the following links give 404s (don't redirect properly):
- https://handsontable.com/docs/11.1/internationalization-i18n
- https://handsontable.com/docs/11.0/internationalization-i18n
- https://handsontable.com/docs/10.0/internationalization-i18n
- https://handsontable.com/docs/9.0/internationalization-i18n
- https://handsontable.com/docs/11.1/keyboard-navigation
- https://handsontable.com/docs/11.0/keyboard-navigation
- https://handsontable.com/docs/10.0/keyboard-navigation
- https://handsontable.com/docs/9.0/keyboard-navigation

### PR DESCRIPTION
**Pull request by kirszenbaum**
This PR changes some of the docs redirects, to:
- Fix #9454
- Limit the amount of maintenance needed

| Link on Production | On Production, currenty redirects to: | After this PR, should redirect to: |
| --- | --- | --- |
| https://handsontable.com/docs/internationalization-i18n | https://handsontable.com/docs/language/ | https://handsontable.com/docs/language/ (should stay that way) |
| https://handsontable.com/docs/11.1/internationalization-i18n | https://handsontable.com/docs/11.1/language/ (404) | https://handsontable.com/docs/11.1/internationalization-i18n/ (there should be no redirect) |
| https://handsontable.com/docs/11.0/internationalization-i18n | https://handsontable.com/docs/11.0/language/ (404) | https://handsontable.com/docs/11.0/internationalization-i18n (there should be no redirect)  |
| https://handsontable.com/docs/10.0/internationalization-i18n | https://handsontable.com/docs/10.0/language/ (404) | https://handsontable.com/docs/10.0/internationalization-i18n/ (there should be no redirect) |
| https://handsontable.com/docs/9.0/internationalization-i18n | https://handsontable.com/docs/9.0/language/ (404) | https://handsontable.com/docs/9.0/internationalization-i18n (there should be no redirect) |
| https://handsontable.com/docs/keyboard-navigation | https://handsontable.com/docs/keyboard-shortcuts/ | https://handsontable.com/docs/keyboard-shortcuts/ (should stay that way) |
| https://handsontable.com/docs/11.1/keyboard-navigation | https://handsontable.com/docs/11.1/keyboard-shortcuts/ (404) | https://handsontable.com/docs/11.1/keyboard-navigation/ (there should be no redirect) |
| https://handsontable.com/docs/11.0/keyboard-navigation | https://handsontable.com/docs/11.0/keyboard-shortcuts/ (404) | https://handsontable.com/docs/11.0/keyboard-navigation/ (there should be no redirect) |
| https://handsontable.com/docs/10.0/keyboard-navigation | https://handsontable.com/docs/10.0/keyboard-shortcuts/ (404) | https://handsontable.com/docs/10.0/keyboard-navigation/ (there should be no redirect) |
| https://handsontable.com/docs/9.0/keyboard-navigation | https://handsontable.com/docs/9.0/keyboard-shortcuts/ (404) | https://handsontable.com/docs/9.0/keyboard-navigation/ (there should be no redirect) |
| https://handsontable.com/docs/hello-world | https://handsontable.com/docs/demo | https://handsontable.com/docs/demo (should stay that way) |
| https://handsontable.com/docs/11.1/hello-world | https://handsontable.com/docs/11.1/demo | (404, there should bo no redirect) |
| https://handsontable.com/docs/11.0/hello-world | https://handsontable.com/docs/11.0/ | (404, there should bo no redirect) |
| https://handsontable.com/docs/10.0/hello-world | https://handsontable.com/docs/10.0/ | (404, there should bo no redirect) |
| https://handsontable.com/docs/9.0/hello-world | https://handsontable.com/docs/9.0/ | (404, there should bo no redirect) |

[skip changelog]

### CODE REVIEW

File: docs/docker/redirects.conf [hunk 1]
```diff
@@ -64,10 +64,14 @@ rewrite ^/((\d+\.\d+|next)/)?tutorial-performance-tips/?$ /docs/$1performance/ p
...
 rewrite ^/((\d+\.\d+|next)/)?tutorial-migration-guide/?$ /docs/$1migration-from-7.4-to-8.0/ permanent;
 rewrite ^/((\d+\.\d+|next)/)?tutorial-known-limitations/?$ /docs/$1third-party-licenses/ permanent;
-rewrite ^/((\d+\.\d+|next)/)?keyboard-navigation/?$ /docs/$1keyboard-shortcuts/ permanent;
-rewrite ^/((\d+\.\d+|next)/)?internationalization-i18n/?$ /docs/$1language/ permanent;
+rewrite ^/keyboard-navigation/?$ /docs/keyboard-shortcuts/ permanent;
+rewrite ^/(12\.0|next)/keyboard-navigation/?$ /docs/$1/keyboard-shortcuts/ permanent;
+rewrite ^/(11\.1|11\.0|10\.0|9\.0)/keyboard-shortcuts/?$ /docs/$1/keyboard-navigation/ permanent;
+rewrite ^/internationalization-i18n/?$ /docs/language/ permanent;
+rewrite ^/(12\.0|next)/internationalization-i18n/?$ /docs/$1/language/ permanent;
+rewrite ^/(11\.1|11\.0|10\.0|9\.0)/language/?$ /docs/$1/internationalization-i18n/ permanent;
```
Comment (warpech) [inline docs/docker/redirects.conf:72]: The above rewrites do not match the table in the PR description. 

Also, how did you test it? It seems that the `$1` is inserted wrongly, comparing to other rules e.g. on line 66.

**Comment by warpech** [comment]
The suggested way to test this is to use `npm run docs:preview...` link presented above.

Possibly, @budnix could present steps how to test it using Ahrefs on the staging environment.

**Review by budnix — COMMENTED** [review]
Yesterday, I fixed those rules. I tested them today on staging with incognito mode and it works. However, the table in the description should be updated to:

| Link | Should redirect to |
| --- | --- |
| https://dev.handsontable.com/docs/11.1/language/ | https://dev.handsontable.com/docs/11.1/internationalization-i18n/ |
| https://dev.handsontable.com/docs/11.0/language/ | https://dev.handsontable.com/docs/11.0/internationalization-i18n/ |
| https://dev.handsontable.com/docs/10.0/language/ | https://dev.handsontable.com/docs/10.0/internationalization-i18n/ |
| https://dev.handsontable.com/docs/9.0/language/ | https://dev.handsontable.com/docs/9.0/internationalization-i18n/ |
| https://dev.handsontable.com/docs/11.1/keyboard-shortcuts/ | https://dev.handsontable.com/docs/11.1/keyboard-navigation/ |
| https://dev.handsontable.com/docs/11.0/keyboard-shortcuts/ | https://dev.handsontable.com/docs/11.0/keyboard-navigation/ |
| https://dev.handsontable.com/docs/10.0/keyboard-shortcuts/ | https://dev.handsontable.com/docs/10.0/keyboard-navigation |
| https://dev.handsontable.com/docs/9.0/keyboard-shortcuts/ | https://dev.handsontable.com/docs/9.0/keyboard-navigation/ |

And based on the rules the new docs versions also has some redirects

| Link | Should redirect to |
| --- | --- |
| https://dev.handsontable.com/docs/12.0/internationalization-i18n/ | https://dev.handsontable.com/docs/12.0/language/ |
| https://dev.handsontable.com/docs/12.0/keyboard-navigation/ | https://dev.handsontable.com/docs/12.0/keyboard-shortcuts/ |

The question is if the above redirects are necessary as it can be a nightmare to maintain them @warpech?

**Review by aninde — APPROVED** [review]
I checked on incognito mode links from [OP](https://github.com/handsontable/handsontable/pull/9453#issue-1226421264) and they works well. 
Links mentioned by @budnix in the https://github.com/handsontable/handsontable/pull/9453#pullrequestreview-964263169 have 404 also in previous versions, e.g. https://handsontable.com/docs/8.0.0/language/.

 I don't understand why you want to correct them, do we use them anywhere?

**Comment by kirszenbaum** [comment]
@budnix @aninde @warpech sorry for the confusion. I now corrected the descriptions of the issue (#9454) and the PR.

**Comment by aninde** [comment]
I still don't understand the description of the PR. It's  not clear to me what 404 means in parentheses in the column `Now redirects to`

Especially since it is inconsistent. Some of the links from this PR are corrected and do not lead to 404, and some still do.

All links from the table in OP lead to correct pages, not as it says 404.
**Except (notice that two last one are regression IMO):**

https://handsontable.com/docs/11.1/language  No redirect (404) 
https://handsontable.com/docs/11.1/internationalization-i18n/
https://handsontable.com/docs/11.0/language No redirect (404) https://handsontable.com/docs/11.0/internationalization-i18n/
https://handsontable.com/docs/10.0/language No redirect (404) https://handsontable.com/docs/10.0/internationalization-i18n/
https://handsontable.com/docs/9.0/language No redirect (404) https://handsontable.com/docs/9.0/internationalization-i18n/
https://handsontable.com/docs/11.1/keyboard-shortcuts No redirect (404) https://handsontable.com/docs/11.1/keyboard-navigation/
https://handsontable.com/docs/11.0/keyboard-shortcuts No redirect (404) https://handsontable.com/docs/11.0/keyboard-navigation/
https://handsontable.com/docs/10.0/keyboard-shortcuts No redirect (404) https://handsontable.com/docs/10.0/keyboard-navigation/
https://handsontable.com/docs/9.0/keyboard-shortcuts No redirect (404)
https://handsontable.com/docs/12.0/internationalization-i18n/ | https://handsontable.com/docs/language/ | https://handsontable.com/docs/language/ (should stay that way)
https://handsontable.com/docs/12.0/keyboard-navigation/



Screenshot 2022-05-06 at 11 25 43

Screenshot 2022-05-06 at 11 26 01

**Review by aninde — CHANGES_REQUESTED** [review]
https://github.com/handsontable/handsontable/pull/9453#issuecomment-1119436298

File: docs/docker/redirects.conf (hunk 1 above)
Comment (kirszenbaum) [reply docs/docker/redirects.conf:72]: https://github.com/handsontable/handsontable/pull/9453#issuecomment-1119386789

**Review by aninde — APPROVED** [review]
👍

**Review by budnix — APPROVED** [review]
👍

**Pull request merged**